pallets/registrar/src/mock.rs

```diff
@@ -148,6 +148,7 @@ impl pallet_registrar::Config for Test {
     type DepositAmount = DepositAmount;
     type RuntimeHoldReason = RuntimeHoldReason;
     type RegistrarHooks = Mock;
+    type InnerRegistrar = ();
```

Comment:
maybe it would be worth setting this to a mock that we can test @Agusrodri , what do you think? something that updates a variable that we can verify through the test for instance

Reply:
Mmm not sure how could we do that to be honest. I set the empty tuple here as this mock is supposed to reflect the standard behavior of the pallet (as its usage in dancebox and flashbox).

In case we would like this type to be dynamic, we could try introducing the `pallet_parameters` to the mock somehow, but still not sure about the implementation. 

If we want to investigate that, I think it would worth doing it in a separate PR, as depending on the final implementation, we could end up having a lot of extra code. What do you think?

Reply:
I think a way to test it is:
- you can have a wrapper type that can be set to () or to something where at least a variable is updated. The goal is to at least know those funcions are being called succesfully

Reply:
At the end I followed another approach that I think it's better. Basically I just leveraged the existing `mock_data` pallet to also include and verify the calls to `InnerRegistrar`.

I updated some tests of `register` and `deregister` where I added extra checks to see if the `Mock` storage of `mock_data` pallet is being updated properly.

The changes are included in 29bf2d8365d566de33bb360f6ec4a98af15f0888. Let me know if this approach works for you, otherwise we can go with the wrapper one :)

Reply:
very nice Agustin (and thanks @tmpolaczyk for inspiring this).

Fixed